jbpm-human-task/jbpm-human-task-core/src/test/java/org/jbpm/services/task/EmailDeadlinesBaseTest.java

```diff
@@ -141,22 +144,80 @@ public void testDelayedEmailNotificationOnDeadline() throws Exception {
         assertTrue(list.contains("new-email@domain.com"));
         assertTrue(list.contains("darth@domain.com"));
     }
     
+    
+    @Test(timeout = 10000)
+    public void testDelayedEmail2NotificationsOnDeadline() throws Exception {  
+        CountDownTaskEventListener countDownListener = new CountDownTaskEventListener(2, false, true);
+        addCountDownListner(countDownListener);
+        
+        Map<String, Object> vars = new HashMap<String, Object>();
+        vars.put("now", new Date());
+
+        Reader reader = new InputStreamReader(getClass().getResourceAsStream(MvelFilePath.DeadlineWith2Notifications));
+        Task task = TaskFactory.evalTask(reader, vars);
+        
+        taskService.addTask(task, new HashMap<String, Object>());
+        long taskId = task.getId();
+
+        InternalContent content = (InternalContent) TaskModelProvider.getFactory().newContent();
+        
+        Map<String, String> params = fillMarshalSubjectAndBodyParams();
+        ContentData marshalledObject = ContentMarshallerHelper.marshal(task, params, null);
+        content.setContent(marshalledObject.getContent());
+        taskService.addContent(taskId, content);
+        long contentId = content.getId();
+        
+        content = (InternalContent) taskService.getContentById(contentId);
+        Object unmarshallObject = ContentMarshallerHelper.unmarshall(content.getContent(), null);
+        checkContentSubjectAndBody(unmarshallObject);
+
+        // emails should not be set yet
+        assertEquals(0, getWiser().getMessages().size());
+        
+        countDownListener.waitTillCompleted();
+        
+        for (WiserMessage msg : getWiser().getMessages()) {
+            logger.info(msg.getEnvelopeReceiver());
+        }
+         assertEquals(2, getWiser().getMessages().size());
+
+        List<String> list = new ArrayList<String>(2);
+        list.add(getWiser().getMessages().get(0).getEnvelopeReceiver());
+        list.add(getWiser().getMessages().get(1).getEnvelopeReceiver());
+        assertTrue(list.contains("new-email@domain.com"));
+        assertTrue(list.contains("darth@domain.com"));
+
+         
+        assertTrue("Cannot find expected message", getWiser().getMessages().stream().filter(m -> {
+            try {
+                MimeMessage msg = m.getMimeMessage();
+                return "new-email@domain.com".equals(((InternetAddress) msg.getRecipients(RecipientType.TO)[0])
+                        .getAddress()) && myBody.equals(msg.getContent()) && mySubject.equals(msg.getSubject()) &&
+                       "from@domain.com".equals(((InternetAddress) msg.getFrom()[0]).getAddress()) &&
+                       "replyTo@domain.com".equals(((InternetAddress) msg.getReplyTo()[0]).getAddress());
+            } catch (MessagingException | IOException e) {
+                throw new IllegalStateException(e);
+            }
+        }).findAny().isPresent());
```

Comment:
you can use `anyMatch` instead, but up to you
```suggestion
        assertTrue("Cannot find expected message", getWiser().getMessages().stream().anyMatch(m -> {
            try {
                MimeMessage msg = m.getMimeMessage();
                return "new-email@domain.com".equals(((InternetAddress) msg.getRecipients(RecipientType.TO)[0])
                        .getAddress()) && myBody.equals(msg.getContent()) && mySubject.equals(msg.getSubject()) &&
                       "from@domain.com".equals(((InternetAddress) msg.getFrom()[0]).getAddress()) &&
                       "replyTo@domain.com".equals(((InternetAddress) msg.getReplyTo()[0]).getAddress());
            } catch (MessagingException | IOException e) {
                throw new IllegalStateException(e);
            }
        }));
```